core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -194,16 +298,51 @@ public String manifestListLocation() {
     return manifestListLocation;
   }
 
+  private void cacheChanges(FileIO fileIO) {
```

Comment:
As with `cacheManifests`, can we combine the implementations so that it is clear there are no changes other than passing in `fileIO`?

Reply:
Done.

Reply:
I think changing the signature is better in case people are using this method in a fork somehow. They'll be more likely to notice the change and update their (additional) code as needed.

The likelihood somebody is using this particular `private` method in some unique way though is likely pretty low in my opinion.

Reply:
Done.